## Functions

- A function is used to package some code in a way that is easy to test and re-use.
- You have already used functions.
  - Some functions, like "print" are built-in functions.
  - But you can make your own functions.

In [1]:
# Function example:

def celsiusToFahrenheit(temp_in_c):
    temp_in_f = (9 / 5) * temp_in_c + 32
    return temp_in_f

In [2]:
c = 100.0
f = celsiusToFahrenheit(c)
print(c, 'C is', f, 'F') 

c = 0.0
f = celsiusToFahrenheit(c)
print(c, 'C is', f, 'F')

c = -40.0
f = celsiusToFahrenheit(c)
print(c, 'C is', f, 'F')

100.0 C is 212.0 F
0.0 C is 32.0 F
-40.0 C is -40.0 F


## Exercise 18

- Write a function that converts Fahrenheit to Celsius
- Print a Fahrenheit to Celsius conversion table from 32 F to 212 F
- The forumla is:
  - C = 5/9 x (F - 32)
  
- Hints:
  - You will need a for loop using the range function.
  - Parentheses effect the order of arithmetic operations.